File: programmers/level_2/오픈채팅방.py
def solution(record):
    answer = []
    dic = {}
    #dic으로 데이터 뽑기 -> change를 고려하기 위해서
    for s in record:
        s_split = s.split(' ')
        if len(s_split) == 3:
            dic[s_split[1]] = s_split[2]
    
    for s in record:
        s_split = s.split(' ')
        if s_split[0] == 'Enter':
            answer.append('{0}님이 들어왔습니다.'.format(dic[s_split[1]]))
        if s_split[0] == 'Leave':
            answer.append('{0}님이 나갔습니다.'.format(dic[s_split[1]]))
    return answer

File: 코딩테스트_합격자되기_자바스크립트/해시/오픈채팅방.js
function solution(record) {
  const result = [];

  // uuid : nickName hash 생성
  // record를 순회하면 최종적으로 바뀐 nickName이 저장된다.
  const hash = record.reduce((acc, part) => {
    const recordSplit = part.split(' ');
    const action = recordSplit[0];

    // 모두 바꿔주면 안되는거지,,,,
    if(action !== 'Leave') {
      const key = recordSplit[1];
      const value =recordSplit[2];
      acc[key] = value;
    }

    return acc
  }, {});

  for (const rec of record) {
    const recSplit = rec.split(' ');
    const uuid = recSplit[1];
    const action = recSplit[0];

    let message = '';
    // Enter, Leave, Change switch 처리
    switch (action) {
      case "Enter":
        message = `${hash[uuid]}님이 들어왔습니다.`;
        break;
      case "Leave":
        message = `${hash[uuid]}님이 나갔습니다.`;
        break;
      case "Change":
        break;
    }

    if(message) {
      result.push(message);
    }

  }
  return result;
}